test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/ViewModels/ReadmePreviewViewModelTests.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Sdk.TestFramework;
+using Moq;
+using NuGet.PackageManagement.UI.ViewModels;
+using NuGet.VisualStudio.Internal.Contracts;
+using Xunit;
+
+namespace NuGet.PackageManagement.UI.Test.ViewModels
+{
+    [Collection(MockedVS.Collection)]
+    public class ReadmePreviewViewModelTests
+    {
+        [Fact]
+        public void ReadmePreviewViewModel_WithNullPackageFileService_ThrowsArgumentNullException()
+        {
+            Assert.Throws<ArgumentNullException>(() =>
+            {
+                new ReadmePreviewViewModel(null);
+            });
+        }
+
+        [Fact]
+        public void ReadmePreviewViewModel_Defaults()
+        {
+            //Arrange
+            var mockFileService = new Mock<INuGetPackageFileService>();
+
+            //Act
+            var target = new ReadmePreviewViewModel(mockFileService.Object);
+
+            //Assert
+            Assert.False(target.ErrorLoadingReadme);
+            Assert.Equal(string.Empty, target.ReadmeMarkdown);
+            Assert.True(target.CanDetermineReadmeDefined);
+        }
+
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData(" ")]
+        public async Task ReadmePreviewViewModel_LoadReadmeAsync_WithoutReadmeUrl_NoReadmeReturned(string readmeUrl)
```

Comment:
We can leave the ReadmePreviewViewModel part off all the tests since this test class already describes the class being tested. See https://github.com/NuGet/NuGet.Client/blob/dev/docs/coding-guidelines.md#unit-test-method-naming

```suggestion
        public async Task LoadReadmeAsync_WithoutReadmeUrl_NoReadmeReturned(string readmeUrl)
```